packages/compass-aggregations/src/components/stage-editor/stage-editor.jsx

```diff
@@ -87,9 +81,13 @@ class StageEditor extends Component {
     );
     this.completer.version = this.props.serverVersion;
     if (this.props.stageOperator !== prevProps.stageOperator && this.editor) {
-      this.editor.setValue('');
-      this.editor.insertSnippet(this.props.snippet || '');
```

Comment:
Here's where we used to perform the snippet completion. This would also call the `onStageChange` twice I believe, and by doing so cause the stage preview to be recalculated. These changes make it so we explicitly try to run now when the operator changes. This might be something we could flag in the store instead.